src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+---
+layout: guide
+title: PDL Schema
+permalink: /annotation_processor
+excerpt: Documentation of using annotation processor in Rest.Li
+---
+
+# Annotation Processor
+This sections introduces a generic schema annotation processing tool in Rest.Li. Before reading this, it is recommended to be familiar with the concept of PathSpec in Rest.Li. [Insert link here]
+
+## Annotating Rest.Li Schemas
+In Rest.Li, schema writers are able to add annotations to schema fields or the schema itself.
+
+**Example**: annotate on the schema fields
+```pdl
+record UserPersonallyIdentifiableInformation {
+  @data_classification = "MEDIUM"
+  firstName: string,
+  @data_classification = "MEDIUM"
+  lastName: string,
+  @data_classification = "LOW"
+  userId: long,
+  @data_classification = "HIGH"
+  socialSecurityNumber: string,
+}
+```
+
+**Example**: annotate directly on the schema
+```pdl
+@data_classification = "HIGH"
+record UserPersonallyIdentifiableInformation {
+  firstName: string,
+  lastName: string,
+  userId: long,
+  socialSecurityNumber: string,
+}
+```
+Note that these annotations are on the schema level.
+
+The support that Rest.Li provided for processing annotation in Rest.Li schema has been quite limited. These annotations in above examples are later stored as a `DataSchema`'s class attribute called "property". Just as the example shows, both field and the DataSchema can have this "property". We let the user chooses where they want to have the schema annotated and how they want to interpret. There were no preferrabled way regarding how annotation should be written, read and interpreted.
```

Comment:
What does the first sentence mean? It sounds like you're saying support was and still is currently limited. Also, I'd recommend not presenting information about features as a story detailing the old support and the new support. This will confuse new users; why does it matter to them that support _used_ to be one way, but _now_ it's another way? It doesn't provide much of a value for users to know that we previously didn't support this. That's my opinion.

nits:
- `let the user chooses` -> `let the user choose`
- `how they want to interpret` -> `how they want it to be interpreted`
- `no preferrabled way` -> `no preferable ways`
- `written, read and interpreted` -> `written, read, and interpreted`

Reply:
All re-worded